src/app/src/test/kotlin/org/imperial/mrc/hint/database/SnapshotRepositoryTests.kt

```diff
@@ -24,19 +28,43 @@ class SnapshotRepositoryTests {
     @Autowired
     private lateinit var sut: SnapshotRepository
 
+    @Autowired
+    private lateinit var versionRepo: VersionRepository
+
+    @Autowired
+    private lateinit var userRepo: UserLogic
+
     @Autowired
     private lateinit var dsl: DSLContext
 
     private val snapshotId = "sid"
+    private val testEmail = "test.user@test.com"
 
     @Test
-    fun `can save snapshot`() {
+    fun `can save snapshot without version id`() {
         sut.saveSnapshot(snapshotId, null)
 
+        val snapshot = dsl.selectFrom(VERSION_SNAPSHOT)
+                .fetchOne()
+        assertThat(snapshot[VERSION_SNAPSHOT.ID]).isEqualTo(snapshotId)
+
+        val versionId: Int? = snapshot[VERSION_SNAPSHOT.VERSION_ID]
```

Comment:
Oddly, version_id has type Int rather than Int? in the generated db interface - it's a foreign key but a nullable foreign key, and it's quite happy to have nulls in the table, but there was a runtime type exception when comparing to null without this cast..